firedrake/slate/slac/compiler.py

```diff
@@ -59,13 +59,12 @@
 BLASLAPACK_INCLUDE = None
 if not complex_mode:
     if COMM_WORLD.rank == 0:
-        petsc_variables = get_petsc_variables()
-        EIGEN_INCLUDE_DIR = petsc_variables.get("EIGEN_INCLUDE")
+        EIGEN_INCLUDE_DIR = get_eigen_include_dir()
         if EIGEN_INCLUDE_DIR is None:
             raise ValueError("""Could not find Eigen configuration in %s. Did you build PETSc with Eigen?""" % PETSC_ARCH or PETSC_DIR)
-        EIGEN_INCLUDE_DIR = EIGEN_INCLUDE_DIR.lstrip('-I')
         EIGEN_INCLUDE_DIR = COMM_WORLD.bcast(EIGEN_INCLUDE_DIR, root=0)
 
+        petsc_variables = get_petsc_variables()
```

Comment:
Is there a reason why we need to call this function after the `get_eigen_include_dir` function?

Reply:
I moved it to separate the logic of getting `EIGEN_INCLUDE_DIR` (which doesn't require `petsc_variables`) from getting `BLASLAPACK_{LIB,INCLUDE}` (which do require it).

Reply:
Okay